packages/mwp-app-render/src/util/seo/metaTags.test.js

```diff
@@ -0,0 +1,168 @@
+	title: 'mock title',
+	description: 'mock description',
+	keywords: 'mock keywords',
+	baseUrl: 'https://mock-base-url.com',
+	route: '/mock/preview/path',
+	appPath: '/mock/app/path',
+	imageUrl: 'https://www.meetup.com/mock-default-image.jpg',
+	localeCode: 'fr-FR',
+};
+
+describe('generateMetaData', () => {
+	const metaData = generateMetaData(MOCK_META);
+
+	it('should set the title', () => {
+		const titleMeta = metaData.filter(
+			obj => obj.content === MOCK_META.title && obj.property === 'og:title'
+		);
+		expect(titleMeta).toHaveLength(1);
+	});
+
+	it('should set the description', () => {
+		const descriptionMeta = metaData.filter(
+			obj =>
+				obj.content === MOCK_META.description &&
+				(obj.property === 'og:description' || obj.name === 'description')
+		);
+		expect(descriptionMeta).toHaveLength(2);
+	});
+
+	it('should set the keywords', () => {
+		const keywordsMeta = metaData.filter(
+			obj => obj.content === MOCK_META.keywords && obj.name === 'keywords'
+		);
+		expect(keywordsMeta).toHaveLength(1);
+	});
+
+	it('should set the app url', () => {
+		const appUrlMeta = metaData.filter(obj => obj.property === 'al:web:url');
+		expect(appUrlMeta).toHaveLength(1);
+	});
+
+	it('should set the appPath', () => {
+		const appPathMeta = metaData.filter(obj => obj.content === MOCK_META.appPath);
+		expect(appPathMeta).toHaveLength(2);
+	});
+
+	it('should set the url as localized url', () => {
+		const expectedUrl = generateCanonicalUrl(
+			MOCK_META.baseUrl,
+			MOCK_META.localeCode,
+			MOCK_META.route
+		);
+		const ogUrlMeta = metaData.filter(
+			obj => obj.content === expectedUrl && obj.property === 'og:url'
+		);
+		const alUrlMeta = metaData.filter(
+			obj => obj.content === expectedUrl && obj.property === 'al:web:url'
+		);
+		expect(ogUrlMeta).toHaveLength(1);
+		expect(alUrlMeta).toHaveLength(1);
+	});
+
+	describe('open graph meta data', () => {
+		it('should set og:site_name to "Meetup"', () => {
+			const ogSiteName = obj =>
+				obj.content === 'Meetup' && obj.property === 'og:site_name';
+			expect(ogSiteName).toHaveLength(1);
+		});
+		it('should set og:type to "article"', () => {
+			const metaData = generateMetaData(MOCK_META);
+			const ogType = metaData.filter(
+				obj => obj.content === 'article' && obj.property === 'og:type'
+			);
+			expect(ogType).toHaveLength(1);
+		});
+		it('should set og:url', () => {
+			const urlMeta = metaData.filter(obj => obj.property === 'og:url');
+			expect(urlMeta).toHaveLength(1);
+		});
+		it('should set og:image', () => {
+			const ogImage = metaData.filter(
+				obj => obj.content === MOCK_META.imageUrl && obj.property === 'og:image'
+			);
+			expect(ogImage).toHaveLength(1);
+		});
+	});
+	describe('twitter meta data', () => {
+		it('should set twitter:image', () => {
+			const twitterImage = metaData.filter(
+				obj =>
+					obj.content === MOCK_META.imageUrl && obj.property === 'twitter:image'
+			);
+			expect(twitterImage).toHaveLength(1);
+		});
+	});
```

Comment:
_all_ of these tests that use `metaData.filter` could be replaced with a single `expect(metaData).toMatchSnapshot()` test, which would be faster and more robust